fearless/Modules/Staking/Analytics/Rewards/AnalyticsRewardsPresenter.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+import BigInt
+import SoraFoundation
+
+final class AnalyticsRewardsPresenter {
+    weak var view: AnalyticsRewardsViewProtocol?
+    let wireframe: AnalyticsRewardsWireframeProtocol
+    let interactor: AnalyticsRewardsInteractorInputProtocol
+    private let logger: LoggerProtocol?
+    private let viewModelFactory: AnalyticsRewardsViewModelFactoryProtocol
+    private let localizationManager: LocalizationManager
+    private var rewardsData: [SubqueryRewardItemData]?
+    private var selectedPeriod = AnalyticsPeriod.default
+    private var selectedPeriodDiff = 0
```

Comment:
Consider to store Calendar entity and base date here. Also subscribe to day changes to update the screen